docs/md/BasicTutorial.md

```diff
@@ -0,0 +1,282 @@
+            {
+                "input_3": 0,
+                "input_2": 3.0,
+                "input_1": input_file,
+            }
+        ),
+    )
+
+    status: JobStatus = solvers_api.start_job(solver.id, solver.version, job.id)
+    while not status.stopped_at:
+        time.sleep(3)
+        status = solvers_api.inspect_job(solver.id, solver.version, job.id)
+        print("Solver progress", f"{status.progress}/100", flush=True)
+    #
+    # Solver progress 0/100
+    # Solver progress 100/100
+
+    outputs: JobOutputs = solvers_api.get_job_outputs(solver.id, solver.version, job.id)
+
+    print(f"Job {outputs.job_id} got these results:")
+    for output_name, result in outputs.results.items():
+        print(output_name, "=", result)
+
+    #
+    # Job 19fc28f7-46fb-4e96-9129-5e924801f088 got these results:
+    #
+    # output_1 = {'checksum': '859fda0cb82fc4acb4686510a172d9a9-1',
+    # 'content_type': 'text/plain',
+    # 'filename': 'single_number.txt',
+    # 'id': '9fb4f70e-3589-3e9e-991e-3059086c3aae'}
+    # output_2 = 4.0
+
+
+    results_file: File = outputs.results["output_1"]
+    download_path: str = files_api.download_file(file_id=results_file.id)
+    print(Path(download_path).read_text())
+    #
+    # 7
+
+```
+
+The script above
+
+1. Uploads a file ``file_with_number.txt``
+2. Selects version ``2.0.2`` of the ``sleeper``
+3. Runs the ``sleeper`` and provides a reference to the uploaded file and other values as input parameters
+4. Monitors the status of the solver while it is running in the platform
+5. When the execution completes, it checks the outputs
+6. One of the outputs is a file and it is downloaded
+
+
+#### Files
+
+Files used as input to solvers or produced by solvers in the platform are accessible in the **files** section and specifically with the ``FilesApi`` class.
+In order to use a file as input, it has to be uploaded first and the reference used in the corresponding solver's input.
+
+
+```python
+files_api = FilesApi(api_client)
+input_file: File = files_api.upload_file(file="file_with_number.txt")
+
+
+# ...
+
+
+outputs: JobOutputs = solvers_api.get_job_outputs(solver.id, solver.version, job.id)
+results_file: File = outputs.results["output_1"]
+download_path: str = files_api.download_file(file_id=results_file.id)
+
+```
+
+In the snippet above, ``input_file`` is a ``File`` reference to the uploaded file and that is passed as input to the solver. Analogously, ``results_file`` is a ``File`` produced by the solver and that can also be downloaded.
+
+
+#### Solvers, Inputs and Outputs
+
+The inputs and outputs are specific for every solver. Every input/output has a name and an associated type that can be as simple as booleans, numbers, strings ... or more complex as files. You can find this information in the UI under Discover Tab, selecting the service card > More Info > raw metadata. For instance, the ``sleeper`` version ``2.1.1`` has the following ``raw-metadata``:
+
+```json
```

Comment:
I'm not sure this version of the service is available on osparc production?

Reply:
true. I adapted it to 2.0.2.